File: HolzraumCatch/Models/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace HolzraumCatch.Models
{
    public class Project : ModelBase
    {
        [Required]
        [MaxLength(255)]
        public string projectName { get; set; }

        [Required]
        [MaxLength(40)]
        public string projectCode { get; set; }

        [MaxLength(255)]
        public string description { get; set; }

        [MaxLength(140)]
        public string address { get; set; }

        public ICollection<ProjectUser> projectUsers { get; set; }

        public ICollection<ImportJob> imports { get; set; }
    }
}

File: HolzraumCatch/ClientApp/src/app/core/models/project.ts
import { ImportJob } from "./import-job";
import { ProjectUser } from "./project-user";
import { SummaryData } from "./summary-data";

export interface Project {
    id: number;
    projectName: string;
    projectCode: string;
    address: string;
    description: string;
    lastUpdated: Date;
    created: Date;

    scopes: SummaryData[];
    functions: SummaryData[];
    slices: SummaryData[];    
    panels: SummaryData[];
    materials: SummaryData[];

    projectUsers: ProjectUser[];

    imports: ImportJob[];
}